weave/wandb_interface/wandb_lite_run.py

```diff
@@ -1,19 +1,45 @@
-import os
-import logging
 import datetime
-import json
+import logging
+import platform
 import typing
+
+from wandb import __version__, util
 from wandb import errors as wandb_errors
-from wandb.apis.public import Run
-from wandb.sdk.internal.file_pusher import FilePusher
-from wandb.sdk.internal import file_stream
-from wandb.sdk.internal.internal_api import Api as InternalApi
-from wandb.sdk.lib import runid
-from weave import wandb_client_api
-from weave import errors
+from wandb.apis import public
+
+from .artifacts import artifact_saver
+from .internal import file_stream
+from .internal.file_pusher import FilePusher
+from .internal.internal_api import Api as InternalApi
+from .internal.thread_local_settings import _thread_local_api_settings
+from .lib import config_util, runid
+from .lib.ipython import _get_python_type
+
+if typing.TYPE_CHECKING:
+    from wandb.sdk.artifacts.artifact import Artifact
 
 logger = logging.getLogger(__name__)
 
+_p_api = None
+
+
+def wandb_public_api() -> public.Api:
+    global _p_api
+    if _p_api is None:
+        _p_api = public.Api(timeout=30)
+    return _p_api
+
+
+def assert_wandb_authenticated() -> None:
+    authenticated = (
+        _thread_local_api_settings.cookies is not None
+        or wandb_public_api().api_key is not None
```

Comment:
Switched the order here because instantiating the `wandb_public_api` checks for an api key on init.

Reply:
same here, why copy?